dev/TEAL.md

```diff
@@ -625,6 +655,28 @@ Account fields used in the `acct_params_get` opcode.
 | `log` | write A to log state of the current application |
 | `block f` | field F of block A. Fail unless A falls between txn.LastValid-1002 and txn.FirstValid (exclusive) |
 
+### Box Access
+
+All box related opcodes fail immediately if used in a
+ClearStateProgram. This behavior is meant to discourage Smart Contract
+authors from depending upon the availability of boxes in a ClearState
+transaction, as accounts using ClearState are under no requirement to
+furnish appropriate Box References.  Authors would do well to keep the
+same issue in mind with respect to the availability of Accounts,
+Assets, and Apps though State Access opcodes _are_ allowed in
+ClearState programs because the current application and sender account
+are sure to be _available_.
+
+| Opcode | Description |
+| - | -- |
+| `box_create` | create a box named A, of length B. Fail if A is empty or B exceeds 32,768. Returns 0 if A already existed, else 1 |
```

Comment:
Need to say what happens if A exists but has length different from B

Reply:
That is mentioned in the more detailed documentation in TEAL_opcodes.md

Newly created boxes are filled with 0 bytes. `box_create` will fail if the referenced box already exists with a different size. Otherwise, existing boxes are unchanged by `box_create`.